### ISSUE TITLE
dart-style defines global.document which causes systemjs bundler failure

### ISSUE THREAD

**Issue by RobiFerentz** [angular/angular#6929]
I'm having an issue building angular locally. 
The console give me this:
```
'!bundle.js.prod' errored after 17 ms
[12:51:37] TypeError: document.getElementsByTagName is not a function
    at /Users/robertferentz/Work/angular/node_modules/systemjs-builder/node_modules/systemjs/dist/system.src.js:2332:28
    at /Users/robertferentz/Work/angular/node_modules/systemjs-builder/node_modules/systemjs/dist/system.src.js:2403:3
    at Object.<anonymous> (/Users/robertferentz/Work/angular/node_modules/systemjs-builder/node_modules/systemjs/dist/system.src.js:2405:3)
    at Module._compile (module.js:397:26)
    at Object.Module._extensions..js (module.js:404:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (/Users/robertferentz/Work/angular/node_modules/systemjs-builder/lib/builder.js:2:14)
[12:51:37] 'build.dart' errored after 2.32 min
[12:51:37] Error: build sequence failed
[12:51:37] 'build/packages.dart' errored after 2.32 min
[12:51:37] Error: build sequence failed
[12:51:37] 'build.js.dev' errored after 2.28 min
[12:51:37] Error: build sequence failed
[12:51:37] 'build/tree.dart' errored after 2.22 min
[12:51:37] Error: build sequence failed
[12:51:37] 'build/tree.dart' errored after 2.22 min
[12:51:37] Error: build sequence failed
[12:51:37] 'build.js.dev' errored after 2.28 min
[12:51:37] Error: build sequence failed
[12:51:37] 'build/tree.dart' errored after 2.22 min
[12:51:37] Error: build sequence failed
[12:51:37] 'build/packages.dart' errored after 2.32 min
[12:51:37] Error: build sequence failed
[12:51:37] 'build.js.dev' errored after 2.28 min
[12:51:37] Error: build sequence failed
[12:51:37] 'build/tree.dart' errored after 2.22 min
[12:51:37] Error: build sequence failed
[12:51:39] Finished 'build.css.material' after 2.19 s
[12:51:40] Finished 'benchpress.bundle' after 50 s


```

Seems to only be an issue after updating the local repo from upstream and then using node 5.4.1 (as required by .nvmrc)

**Comment by gbataille** [comment]
+1

**Comment by ericmartinezr** [comment]
I'm seeing the same error message (using Node 5.5.0). I workaround it by taking a couple of lines of the latest systemjs (0.19.22). The line 2332 is this one

```
  if (typeof document != 'undefined') {
    var head;

    var scripts = document.getElementsByTagName('script'); // Line 2332
```
clearly the if passes and it crashes the building. So I changed that if to what we see in this line https://github.com/systemjs/systemjs/blob/master/dist/system.src.js#L1405

Ending up in 
```
  var isBrowser = typeof window != 'undefined' && typeof document != 'undefined';
  if (isBrowser && document.getElementsByTagName) {
```
And with that it passes BUT after that it throws another error :(
```
Error: Unable to calculate path for file:/home/eric/html/angular/dist/js/prod/es5/rxjs/CoreOperators.js
```
The file exists, so I don't know what's going on there and the building process fails once again :(

**Comment by alexeagle** [comment]
We made a fix for this at Victor's desk, he should have a PR coming.

**Comment by hterkelsen** [comment]
+1! This is blocking me from building angular

**Comment by vikerman** [comment]
This should be fixed once #7195 hits master - which should be soon.

**Comment by vikerman** [comment]
#7195 is in master now.
